app/controllers/api/vms_controller.rb

```diff
@@ -228,6 +228,16 @@ def request_console_resource(type, id = nil, data = nil)
       end
     end
 
+    def assign_tags_resource(type, id, data)
+      vm = resource_search(id, type, collection_class(:vms))
+      tag_results = data['tags'].collect do |tag|
+        tags_assign_resource(vm, type, tag['id'], tag)
+      end
+      { id => tag_results }
+    rescue => err
+      { id => action_result(false, err.to_s) }
+    end
```

Comment:
See if this can't be generalized, will be easier to implement assign_tags/unassign_tags to all other collections that support the tags subcollection.